tests/test_raster.py

```diff
@@ -108,6 +108,17 @@ def test_arr(self):
         self.assertEqual(img_last.arr.shape, (385, 502, 1))
         img_last.close()
 
+    def test_set_array(self):
+        img = Image(TEST_FILE, dimorder="first")
+        img.arr = img.arr + 1
+
+        self.assertEqual(np.sum(img.arr), np.sum(np.ones(shape=img.arr.shape)))
+
+    def test_set_array_error(self):
+        with self.assertRaises(TypeError):
+            img = Image(TEST_FILE, dimorder="first")
+            img.arr = "error"

```

Comment:
better test with an array of another shape

Reply:
I would check for both.